sw/airborne/modules/computer_vision/opencv_contour_RM.cpp

```diff
@@ -0,0 +1,209 @@
+printf("Detected %d lines\n",lines.size());
+/* // compute the intersection from the lines detected...
+vector<Point2f> intersections;
+for( size_t i = 0; i < lines.size(); i++ )
+{
+for(size_t j = 0; j < lines.size(); j++)
+{
+Vec2f line1 = lines[i];
+Vec2f line2 = lines[j];
+if(acceptLinePair(line1, line2, CV_PI / 32))
+{
+Point2f intersection = computeIntersect(line1, line2);
+intersections.push_back(intersection);
+}
+}
+}
+if(intersections.size() > 0)
+{
+vector<Point2f>::iterator i;
+for(i = intersections.begin(); i != intersections.end(); ++i)
+{
+cout << "Intersection is " << i->x << ", " << i->y << endl;
+circle(occludedSquare, *i, 1, Scalar(0, 255, 0), 3);
+}
+}*/
+/// Find contours
+Mat B = gray.clone();
+findContours( B, contours, hierarchy, CV_RETR_TREE, CV_CHAIN_APPROX_SIMPLE, Point(0, 0) );
+/// Draw contours
+Mat drawing = Mat::zeros( canny_output.size(), CV_8UC3 );
+// printf("Found %d contours\n",contours.size());
+vector<vector<Point> > squareContours;
+for( int i = 0; i< contours.size(); i++ )
+{
+vector<Point> approx;
+approxPolyDP(contours[i], approx, 5, true);
+// printf("Length approx: %d\n",approx.size());
+if(approx.size()==4 ){
+// if( isContourConvex(contours[i])){
+Scalar color = Scalar( rng.uniform(0, 255), rng.uniform(0,255), rng.uniform(0,255) );
+drawContours( gray, contours, i, color, 2, 8, hierarchy, 0, Point() );
+}
+}
+// display
+// cv::imshow("input", input);
+// cv::imshow("drawing", drawing);
+// cv::waitKey(0);
+//
+// cv::imwrite("rotatedRect.png",drawing);
+// cv::cvtColor(drawing,image, CV_GRAY2RGB);
+// Turn the opencv RGB colored image back in a YUV colored image for the drone
+// for (int row = 0; row < height; row++) {
+// for (int col = 0; col < width; col++) {
+// cv::Vec3b pixelHere = image.at<cv::Vec3b>(row, col);
+// img[(row * width + col) * 2 + 1] = 0.299 * pixelHere[0] + 0.587 * pixelHere[1] + 0.114 * pixelHere[2];
+// if (col % 2 == 0) { // U
+// img[(row * width + col) * 2] = 0.492 * (pixelHere[2] - img[(row * width + col) * 2 + 1] + 127);
+// } else { // V
+// img[(row * width + col) * 2] = 0.877 * (pixelHere[0] - img[(row * width + col) * 2 + 1] + 127);
+// }
+// }
+// }
+// void convertBackFromRGB(Mat image, uint8_t* img, int width, int height){
+//
+// //Turn the opencv RGB colored image back in a YUV colored image for the drone
+// for (int row = 0; row < height; row++) {
+// for (int col = 0; col < width; col++) {
+// cv::Vec3b pixelHere = image.at<cv::Vec3b>(row, col);
+// img[(row * width + col) * 2 + 1] = 0.299 * pixelHere[0] + 0.587 * pixelHere[1] + 0.114 * pixelHere[2];
+// if (col % 2 == 0) { // U
+// img[(row * width + col) * 2] = 0.492 * (pixelHere[2] - img[(row * width + col) * 2 + 1] + 127);
+// } else { // V
+// img[(row * width + col) * 2] = 0.877 * (pixelHere[0] - img[(row * width + col) * 2 + 1] + 127);
+// }
+// }
+// }
+// }
+//
+convertBackFromGrayscale( gray, img, width, height);
+//convertBackFromRGB( image, img, width, height);
+return 0;
+}
```

Comment:
there are no tabs in this file, please fun ./fix_code_style on all the files you commit